schemas/x509.yml

```diff
@@ -0,0 +1,171 @@
+    This implements the common core fields for x509 certificates. This information is likely logged with TLS sessions,
+    digital signatures found in executable binaries, S/MIME information in email bodies, or analysis of files on disk.
+    When only a single certificate is logged in an event, it should be nested under `file`. When hashes of the DER-encoded
+    certificate are available, the `hash` data set should be populated as well (e.g. `file.hash.sha256`). For events that
+    contain certificate information for both sides of the connection, the x509 object could be nested under the respective
+    side of the connection information (e.g. `tls.server.x509`).
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - file
+      - tls.client
+      - tls.server
+  fields:
+    - name: version_number
+      level: extended
+      type: keyword
+      description: Version of x509 format.
+      example: 3
+
+    - name: serial_number
+      level: extended
+      type: keyword
+      short_description: Unique serial number issued by the certificate authority.
+      description: >
+        Unique serial number issued by the certificate authority. For consistency, if this value is alphanumeric, it should be
+        formatted without colons and uppercase characters.
+      example: 55FBB9C7DEBF09809D12CCAA
+
+    - name: issuer.distinguished_name
+      level: extended
+      type: keyword
+      description: Distinguished name (DN) of issuing certificate authority.
+      example: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert SHA2 High Assurance Server CA
+
+    - name: issuer.common_name
+      level: extended
+      type: keyword
+      description: Common name (CN) of issuing certificate authority.
+      example: DigiCert SHA2 High Assurance Server CA
+
+    - name: issuer.organizational_unit
+      level: extended
+      type: keyword
+      normalize:
+        - array
+      description: List of organizational units (OU) of issuing certificate authority.
+      example: www.digicert.com
+
+    - name: issuer.organization
+      level: extended
+      type: keyword
+      description: Organizational units (O) of issuing certificate authority.
+      example: DigiCert Inc
+
+    - name: issuer.locality
+      level: extended
+      type: keyword
+      description: Locality name (L)
+      example: Mountain View
+
+    - name: issuer.state_or_province
+      level: extended
+      type: keyword
+      description: State or province name (ST, S, or P)
+      example: California
+
+    - name: issuer.country
+      level: extended
+      type: keyword
+      description: Country (C) code
+      example: US
+
+    - name: signature_algorithm
+      level: extended
+      type: keyword
+      description: Identifier for certificate signature algorithm. Recommend using names found in Go Lang Crypto library (https://golang.org/src/crypto/x509/x509.go).
+      example: SHA256-RSA
+
+    - name: valid_date
```

Comment:
I find the naming here, `valid_date` and `expire_date`, a bit confusing. I'm no expert, but I've always seen these dates referred to as `not_before` and `not_after`, the naming used in rfc5280. Is there a reason to deviate from (afaik) the expected names?

Reply:
It was after this suggestion https://github.com/elastic/ecs/pull/762#pullrequestreview-381404470

I don't have a strong opinion either way. The current wording `valid_date` and `expire_date` are a bit less surprising to newcomers to certificates. But you're right that everywhere else they're called `not_before` and `not_after`.

In this case the official wording is surprising to newcomers, but not actually misleading. So I'd also be in favour of using the RFC wording.

There's other cases where the conventional naming is actively confusing to users (e.g. Linux `ctime`, which does **not** mean creation time). In cases of the convention being actively confusing, I'd be ok with overriding and picking a better name.

But my vote is for the quirky `not_before` and `not_after` here. I'm open to hearing other points of view, though.

Reply:
ECS has the existing fields ``tls.server.not_before`` and ``tls.server.not_after`` so I think consistency is better here.

On a related note, what happens with the fields that are already in ``tls.server.*`` and ``tls.client.*`` that overlap? Is the intention that implementations should copy them for now? I think many of the broken out fields like issuer are useful even if not every part of ``x509.*`` is populated. In the existing ECS fields that's currently a leaf node with the distinguished name.

Reply:
happy to change this back and consistency is certainly important. I'll undo this change